apps/studio/src/features/users/components/UserTable/UserTable.tsx

```diff
@@ -0,0 +1,148 @@
+import { useContext, useMemo } from "react"
+import { Text, VStack } from "@chakra-ui/react"
+import {
+  createColumnHelper,
+  getCoreRowModel,
+  getPaginationRowModel,
+  useReactTable,
+} from "@tanstack/react-table"
+
+import type { UserTableData } from "./types"
+import { TableHeader } from "~/components/Datatable"
+import { Datatable } from "~/components/Datatable/Datatable"
+import { UserManagementContext } from "~/features/users"
+import { useTablePagination } from "~/hooks/useTablePagination"
+import { trpc } from "~/utils/trpc"
+import { LastLoginCell } from "./LastLoginCell"
+import { UserTableMenu } from "./UserTableMenu"
+
+export interface UserTableProps {
+  siteId: number
+  getIsomerAdmins: boolean
+}
+
+const columnsHelper = createColumnHelper<UserTableData>()
+
+const getColumns = ({
+  siteId,
+  shouldShowActions,
+}: Pick<UserTableProps, "siteId"> & { shouldShowActions: boolean }) => {
+  const baseColumns = [
+    columnsHelper.display({
+      id: "user_info",
+      header: () => <TableHeader>Collaborator</TableHeader>,
+      cell: ({ row }) => (
+        <VStack gap="0.25rem" align="start">
+          <Text textStyle="subhead-2">{row.original.name}</Text>
+          <Text textStyle="caption-2" textColor="base.content.medium">
+            {row.original.email}
+          </Text>
+        </VStack>
+      ),
+    }),
+    columnsHelper.display({
+      id: "user_role",
+      header: () => <TableHeader>Role</TableHeader>,
+      cell: ({ row }) => <Text textStyle="caption-2">{row.original.role}</Text>,
+      size: 80,
+    }),
+    columnsHelper.display({
+      id: "user_last_login",
+      header: () => <TableHeader>Last login</TableHeader>,
+      cell: ({ row }) => (
+        <LastLoginCell lastLoginAt={row.original.lastLoginAt} />
+      ),
+      size: 80,
+    }),
+  ]
+
+  if (!shouldShowActions) {
+    return baseColumns
+  }
+
+  return [
+    ...baseColumns,
+    columnsHelper.display({
+      id: "user_menu",
+      header: () => <TableHeader>Actions</TableHeader>,
+      cell: ({ row }) => (
+        <UserTableMenu
+          siteId={siteId}
+          userId={row.original.id}
+          userName={row.original.name}
+        />
+      ),
+      size: 24,
+    }),
+  ]
+}
+
+export const UserTable = ({ siteId, getIsomerAdmins }: UserTableProps) => {
```

Comment:
[nit] could we rename `getIsomerAdmins` to `shouldRetrieveIsomerAdmins`? the naming here makes me think it's a function

Reply:
updated to https://github.com/opengovsg/isomer/pull/1136/commits/14259404a7cfdba8024b1153d0fc51f29ee6c0ea !